### PR TITLE
Amb-2 create ci cd pipeline

### PR DESCRIPTION
**Pull request by stolarskimateusz**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [flake8]
 max-line-length=120
-exclude = .git,__pycache__,dist
\ No newline at end of file
+exclude = .git,__pycache__,specification-release/apigee_client.py,specification-release/spec_uploader.py,dist
```
Comment (Sourceless) [inline .flake8:3]: If this is using deploy-service then you'll likely not need this any more

File: Makefile [hunk 2]
```diff
@@ -51,5 +52,17 @@ deploy-spec: update-examples
...
+
+build-proxy:
+	scripts/build_proxy.sh
+
+release: clean publish build-proxy
+	mkdir -p dist
+	tar -zcvf dist/package.tar.gz build
+	cp -r terraform dist
+	cp -r build/. dist
+	#cp -r specification-release dist
```
Comment (Sourceless) [inline Makefile:64]: Remove commented-out line

File: sandbox/config.json [hunk 3]
```diff
@@ -5,21 +5,21 @@
     "latency": 50,
     "logRequestHeaders": false,
     "webServices": {
-        "world": {
+        "hello/world": {
```
Comment (Sourceless) [inline sandbox/config.json:8]: Check in with Tony for what he wants these URLs to be - at the moment it looks like they'd be `hello-world/hello/world`

File: sandbox/mock/helloworld.json [hunk 4]
```diff
@@ -1,3 +1,3 @@
 {
-  "message": "Hello World!"
-}
\ No newline at end of file
+  "message": "Test2!"
```
Comment (Sourceless) [inline sandbox/mock/helloworld.json:2]: I believe the spec defines what these should be (hello world, hello application, hello user) -- make sure it coforms with the spec-generated documentation

File: specification-release/apigee_client.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
+import requests
```
Comment (Sourceless) [inline specification-release/apigee_client.py:1]: This file isn't needed, it's provided by https://github.com/NHSDigital/api-management-utils during release

File: specification/hello-world.yaml [hunk 6]
```diff
@@ -20,7 +20,7 @@ info:
     email: api.management@nhs.net
 
 servers:
-  - url: https://emea-demo8-test.apigee.net
+  - url: https://internal-dev.api.service.nhs.uk
     description: Test Environment
   - url: https://emea-demo8-prod.apigee.net
```
Comment (Sourceless) [inline specification/hello-world.yaml:25]: Remove

File: specification/hello-world.yaml [hunk 7]
```diff
@@ -20,7 +20,7 @@ info:
     email: api.management@nhs.net
 
 servers:
-  - url: https://emea-demo8-test.apigee.net
+  - url: https://internal-dev.api.service.nhs.uk
```
Comment (Sourceless) [inline specification/hello-world.yaml:23]: Again check with Tony what he wants here - I suspect `sandbox.api.service.nhs.uk` is correct here

File: specification/hello-world.yaml [hunk 8]
```diff
@@ -137,7 +137,7 @@ components:
           authorizationUrl: https://emea-demo8-test.apigee.net/oauth2/authorize
```
Comment (Sourceless) [inline specification/hello-world.yaml:137]: These will need updating as well

File: terraform/main.tf [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  backend "azurerm" {}
+
+  required_providers {
+    apigee = "~> 0.0"
+    archive = "~> 1.3"
+  }
+}
+
+module "hello-world" {
+  source             = "github.com/NHSDigital/api-platform-service-module?ref=apm-501-change-proxy-location"
```
Comment (Sourceless) [inline terraform/main.tf:16]: Can remove the ref on this as the branch has since been merged

File: specification-release/apigee_client.py (hunk 5 above)
Comment (stolarskimateusz) [reply specification-release/apigee_client.py]: removed

File: Makefile (hunk 2 above)
Comment (stolarskimateusz) [reply Makefile]: removed

File: .flake8 (hunk 1 above)
Comment (stolarskimateusz) [reply .flake8]: changed

File: terraform/main.tf (hunk 9 above)
Comment (stolarskimateusz) [reply terraform/main.tf]: changed

File: specification/hello-world.yaml (hunk 6 above)
Comment (stolarskimateusz) [reply specification/hello-world.yaml:25]: @Sourceless what should be removed?

File: sandbox/config.json (hunk 3 above)
Comment (stolarskimateusz) [reply sandbox/config.json:8]: Acc. to Tony:
the first level of the resource URL matches the API service name, so for the "Hello World" API, the first level of the resource URL should probably be "/hello-world" i.e. leave it as-is.

File: specification/hello-world.yaml (hunk 7 above)
Comment (stolarskimateusz) [reply specification/hello-world.yaml:23]: @Sourceless you're right

File: specification/hello-world.yaml [hunk 10]
```diff
@@ -20,10 +20,8 @@ info:
     email: api.management@nhs.net
 
 servers:
-  - url: https://internal-dev.api.service.nhs.uk
+  - url: https://sandbox.api.service.nhs.uk
```
Comment (artronics) [inline specification/hello-world.yaml:23]: I think we should have 3 env as in PDS:
```yaml
  - url: 'https://sandbox.api.service.nhs.uk/electronic-prescriptions'
    description: Sandbox environment.
  - url: 'https://int.api.service.nhs.uk/electronic-prescriptions'
    description: Integration test environment.
  - url: 'https://api.service.nhs.uk/electronic-prescriptions'
    description: 'Production environment (not yet available: see "API status and roadmap").'
```

**Pull request closed without merging**